overlord/snapstate/snapstate.go

```diff
@@ -119,6 +119,20 @@ func doInstall(s *state.State, curActive bool, ss *SnapSetup) (*state.TaskSet, e
 	linkSnap := s.NewTask("link-snap", fmt.Sprintf(i18n.G("Make snap %q available to the system%s"), ss.Name, revisionStr))
 	addTask(linkSnap)
 
+	// do GC!
+	if snapst.HasCurrent() {
+		prev := linkSnap
+		seq := snapst.Sequence
+		currentIndex := snapst.findIndex(snapst.Current)
+		for i := 0; i <= currentIndex-2; i++ {
+			si := seq[i]
+			ts := removeInactiveRevision(s, ss.Name, si.Revision)
+			ts.WaitFor(prev)
+			tasks = append(tasks, ts.Tasks()...)
+			prev = tasks[len(tasks)-1]
```

Comment:
just wondering here, this is interesting though given that is the same file not that important, I wondered if we should have a way to attach designated tasks to TaskSets (like Start and End) so that this would become ts.End() or ts.EndTask()